packages/react-core/src/components/Dropdown/InternalDropdownItem.tsx

```diff
@@ -207,6 +227,8 @@ export class InternalDropdownItem extends React.Component<InternalDropdownItemPr
                     ...additionalProps,
                     className: css(classes, itemClass, variant === 'icon' && styles.modifiers.icon)
                   })
+                ) : isRouterLink ? (
```

Comment:
we are already checking if the component is a `isComponentReactElement`. Can we instead update that code path to work for router? Then we might not even need an extra prop